### DISCUSSION TITLE
Parse filename / drop extension

### DISCUSSION THREAD

**Discussion by danabooth** [vlang/v#21522] [Questions and Answers]
Very new to V, (and not an experienced programmer by any means) and just ran into the situation where I'd like to drop the extension from a filename which is a command line argument. Now of course I could do that by hand, but didn't want to make something just to have someone dump on me because there's already a function for that. If there is, Startpage can't find it, but is there?

**Comment by JalonSolov** [comment]
You could use https://modules.vlang.io/index.html#string.all_before_last and pass '.' as the parameter.

If it might be a full path, use https://modules.vlang.io/os.html#file_name first to strip of the name, then all_before_last to strip off the extension.

**Reply by danabooth** [reply]
What I ended up doing is use file_ext to get the extension and use trim_right to drop it. Easy enough, but being new to this, I didn't want the perfect code cops dumping on me.

**Reply by JalonSolov** [reply]
There's no such things as "perfect code".  It either works, or it doesn't.  After that, it's just optimizations.  :-)
